### DISCUSSION TITLE
Compiling in Cygwin

### DISCUSSION THREAD

**Discussion by starik1968** [borgbase/vorta#827] [Packaging]
Hello, try to compile under Cygwin and get this at the end of the compile:

 g++ -shared -o cygQt.dll sipQtcmodule.o  -lpthread
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0xf): undefined reference to `PyErr_Clear'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0xf): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyErr_Clear'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x17): undefined reference to `PyImport_ImportModule'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x17): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyImport_ImportModule'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x27): undefined reference to `PyModule_GetDict'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x27): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyModule_GetDict'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x35): undefined reference to `PyDict_Merge'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x35): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyDict_Merge'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x75): undefined reference to `PyModule_Create2'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x75): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyModule_Create2'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x89): undefined reference to `PyModule_GetDict'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x89): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyModule_GetDict'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x172): undefined reference to `PyErr_Clear'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x172): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `PyErr_Clear'
    /usr/lib/gcc/x86_64-pc-cygwin/10/../../../../x86_64-pc-cygwin/bin/ld: sipQtcmodule.o:sipQtcmodule.c:(.text+0x5c): undefined reference to `_Py_Dealloc'
    sipQtcmodule.o:sipQtcmodule.c:(.text+0x5c): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_Py_Dealloc'
    collect2: Fehler: ld gab 1 als Ende-Status zurück
    make[1]: *** [Makefile:145: cygQt.dll] Fehler 1
    make[1]: Verzeichnis „/tmp/tmp8nr6t6bo/Qt“ wird verlassen
    make: *** [Makefile:65: sub-Qt-make_first-ordered] Fehler 2

And this:

Project ERROR: Unknown module(s) in QT: serialport

Project ERROR: Unknown module(s) in QT: multimedia

 Project ERROR: Unknown module(s) in QT: axcontainer

During comiling of this:

This is the GPL version of PyQt 5.15.1 (licensed under the GNU General Public License) for Python 3.6.10 on cygwin.

Tried now for a couple of days, without success, anybody any hint for me want is missing/went wrong? With wsl2 everything runs well.

Thanks in advance

Best Regards
starik1968

**Comment by m3nu** [comment]
Kudos for trying, @starik1968 ! Can't be of much help on the Qt compile issues, but will pin this issue, so more people notice.

**Comment by rapgro** [comment]
Did you see that thread about similiar linkage issues with boost? Doubtful this issue is directly related to Cygwin, maybe incompatible Cython as a better guess where to begin with debugging, all those Py_methods are coming from Cython. Cython can really be a beast to fight. Do you use the same versions for both Cython and Python?

https://bugs.archlinux.org/task/22800

**Comment by rapgro** [comment]
http://docs.cython.org/en/latest/src/userguide/debugging.html
